core/gui/src/app/dashboard/component/user/user-dataset/user-dataset-explorer/user-dataset-explorer.component.ts

```diff
@@ -213,6 +213,39 @@ export class UserDatasetExplorerComponent implements OnInit {
     }
   }
 
+  onClickDownloadVersionAsZip() {
+    if (this.did && this.selectedVersion && this.selectedVersion.dvid) {
+      this.datasetService
+        .retrieveDatasetVersionZip(this.currentDisplayedFileName)
+        .pipe(untilDestroyed(this))
+        .subscribe({
+          next: blob => {
+            // Create URL for the ZIP blob
+            const url = URL.createObjectURL(blob);
+
+            // Create a temporary link element
+            const a = document.createElement("a");
+            a.href = url;
+            a.download = `${this.datasetName}-${this.selectedVersion?.name}.zip`; // Filename for ZIP file
+
+            // Append the link to the body
+            document.body.appendChild(a);
+            // Trigger the download
+            a.click();
+            // Remove the link after download
+            document.body.removeChild(a);
+            // Release the blob URL
+            URL.revokeObjectURL(url);
```

Comment:
is this the common practice to use a temproary element to download something ?

Reply:
yes, I think it is a common practice, and this is the same as download single file code for this part

Reply:
Ok.